### ISSUE TITLE
Fix Indentation Replaces Tabs with Sapces

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1563]
_From [colly...@gmail.com](https://code.google.com/u/111966524543057858897/) on 2013-09-13T16:19:45Z_

Please note that reporting bugs from Spyder's help menu ("?" > "Report
issue...") will fill version details for Spyder, Python and Qt in this description for you. 

What steps will reproduce the problem?

 1. Change the default behavior of using tabs from using spaces (4) to actual tabs in the preferences.
2. Write some code using tabs
3. Use the Source/Fix Indentation in the top drop down menu and watch your tabs be turned into spaces. What is the expected output? What do you see instead? I would expect the Tabs to remain and any spaces of 4 to be converted to tabs instead of the other way around since I've set tabs as the default in my preferences. What version of the product are you using? On what operating system? Version 2.2.3
Windows 7
From the Help, About :
Python 2.7.5 32bits, Qt 4.8.4 PyQt4 (API v2) 4.9.6 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1563_

**Comment by Anynomouss** [comment]
Using Spyder 5.4.2 now in 2023 and exactly the same issue! 
-Typing a tab in the search and replace box to fix indentation also does not work. 
-Only typing within the Spyder editor, then copying the tab character from the code and  pasting it as replacement (Ctrl +R to bring up the replace menu) for 4 spaces and selecting replace all works to manual fix indentation issues. 
Unbelievable that such a simple problem has not been fixed in the past 7 years. And while we are at it, unimaginable that the use of spaces, which is ambiguous and error prone is still stimulated and set as default by Spyder and advised by PEP8. But lest not start that discussion here. Small flaw in an otherwise awesome editor.

**Comment by ccordoba12** [comment]
@Anynomouss, are you trying to transform a file that uses only tabs for one that uses spaces instead? I say it because (if I'm not mistaken) `Fix indentation` fixes files that have a mix of tabs and spaces, but it doesn't cover the tabs to spaces functionality.

**Comment by Anynomouss** [comment]
@ccordoba12 Yes, that is what I am trying to do. The expected behavior would be to use the Spyders settings for indentation.
So I would expect that a) if indentation is set to spaces, it would fix mixed tab and space indentation to spaces only and b) if Spyder is set to tab indentation, it would fix indentation  to tabs only. 
What surprises me mostly is that this is relatively straight forward to implement, probably only a single line of code to use the Spyder indentation settings instead of a default of 4 spaces....
